ansible/roles/debops.postfix_virtual_ldap/defaults/main.yml

```diff
@@ -0,0 +1,685 @@
+  description: 'Account used by the "Postfix" service to access the LDAP directory'
+
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__ldap_binddn [[[
+#
+# The Distinguished Name of the account LDAP object used by the
+# Postfix service to bind to the LDAP directory.
+postfix_virtual_ldap__ldap_binddn: '{{ ([ postfix_virtual_ldap__ldap_self_rdn ]
+                                        + postfix_virtual_ldap__ldap_device_dn) | join(",") }}'
+
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__ldap_bindpw [[[
+#
+# The password stored in the account LDAP object used by the Postfix service
+# to bind to the LDAP directory.
+postfix_virtual_ldap__ldap_bindpw: '{{ lookup("password", secret + "/ldap/credentials/"
+                                  + postfix_virtual_ldap__ldap_binddn | to_uuid + ".password length=32 "
+                                  + "chars=alpha,digits,!@_#$%^&*") }}'
+
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__ldap_people_rdn [[[
+#
+# The Relative Distinguished Name of the LDAP object which contains the user
+# accounts stored in LDAP.
+postfix_virtual_ldap__ldap_people_rdn: '{{ ansible_local.ldap.people_rdn
+                               if (ansible_local|d() and ansible_local.ldap|d() and
+                                   ansible_local.ldap.people_rdn|d())
+                               else "ou=People" }}'
+
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__ldap_people_dn [[[
+#
+# The Distinguished Name of the LDAP object which contains the user accounts
+# used by Postfix.
+postfix_virtual_ldap__ldap_people_dn: '{{ [ postfix_virtual_ldap__ldap_people_rdn ]
+                                          + postfix_virtual_ldap__ldap_base_dn }}'
+
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__ldap_uri [[[
+#
+# List of LDAP URIs that point to the directory servers which should be used by
+# nextcloudFIXME.
+postfix_virtual_ldap__ldap_uri: '{{ (ansible_local.ldap.uri
+                                    if (ansible_local|d() and ansible_local.ldap|d() and
+                                        ansible_local.ldap.uri|d())
+                                    else [""]) | first }}'
+
+                                                                   # ]]]
+# .. envvar:: postfix_virtual_ldap__ldap_private_groups [[[
+#
+# When this variable is enabled, the :ref:`debops.ldap` role will create
+# a separate LDAP objects that manage the Postfix groups as subtree of the
+# DokiWiki service LDAP object. If you set this parameter to ``False``, the
+# role will use the global ``ou=Groups,dc=example,dc=org`` subtree instead.
+postfix_virtual_ldap__ldap_private_groups: True
```

Comment:
Hello, DokuWiki. :-)

I'm not sure if managing multiple separate mail domains using single LDAP infrastructure is that common, especially that you probably want to have just a single set of `ou=People` objects, which would imply that users share accounts on multiple domains, so having just a single set of mail domains for the entire LDAP tree would suffice. This would also make the configuration a bit simpler... Thoughts?

Reply:
I think it helps if for example you are hosting email for different domains, not subdomains but completely different domains. This works for the use case of and "email provider" outside of and organisation.

Reply:
In that case I would rename the variable to `postfix_virtual_ldap__ldap_private_subtree` to be more precise about the intended purpose, assuming that the "host" subtree would contain more Postfix-related objects than just some groups.

But then again, I suppose that's the intended scalability model. For example, in an environment you might want to deploy a few Postfix instances for round-robin load balancing. Would you expect all instances to use the same set of LDAP objects (horizontal scaling, I think), or each Postfix host to use separate set of users/mailboxes/domains/aliases (vertical scaling)? With how the DebOps project directories are focused on a single domain and DNS MX records support horizontal scaling, I would make use of the global set of LDAP objects rather than private subtree as the default, `False` here. The variable of course can stay so that vertical scaling can easily be enabled.

Reply:
Set new default, updated docs